rc/base/haskell.kak

```diff
@@ -12,19 +12,53 @@ hook global BufCreate .*[.](hs) %{
 # ‾‾‾‾‾‾‾‾‾‾‾‾
 
 add-highlighter -group / regions -default code haskell \
-    string   %{(?<!')"} (?<!\\)(\\\\)*"      ''  \
-    comment  (--[^>])   $                    ''  \
-    comment \{-         -\}                  \{- \
-    macro   ^\h*?\K#    (?<!\\)\n            ''
+    string   '(?<!\'\\)(?<!\')"'         (?<!\\)(\\\\)*" ''   \
+    macro   ^\h*?\K#                     (?<!\\)\n       ''   \
+    pragma  \{-#                         \#-\}           \{-  \
+    comment \{-                            -\}           \{-  \
+    comment --(?![!#$%&*+./<>?@\\\^|~=]) $               ''
 
 add-highlighter -group /haskell/string  fill string
 add-highlighter -group /haskell/comment fill comment
+add-highlighter -group /haskell/pragma  fill meta
 add-highlighter -group /haskell/macro   fill meta
 
-add-highlighter -group /haskell/code regex \b(import)\b 0:meta
-add-highlighter -group /haskell/code regex \b(True|False)\b 0:value
-add-highlighter -group /haskell/code regex \b(as|case|class|data|default|deriving|do|else|hiding|if|in|infix|infixl|infixr|instance|let|module|newtype|of|qualified|then|type|where)\b 0:keyword
-add-highlighter -group /haskell/code regex \b(Int|Integer|Char|Bool|Float|Double|IO|Void|Addr|Array|String)\b 0:type
+add-highlighter -group /haskell/code regex (?<=\W)0x+[A-Fa-f0-9]+ 0:value
```

Comment:
The simpler the better, unless you have a good reason for using the current expression. Also if we ever switch over to `std::regex`, we'll have to get rid of look-behinds since they're not implemented by it.